contracts/L2ApplicationStateRootProvider.sol

```diff
@@ -0,0 +1,17 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity 0.8.25;
+
+/// @title L2ApplicationStateRootProvider
+/// @notice The L2ApplicationStateRootProvider stores the state root of the Cosmos application using Monomer.
+///         This is necessary for implementing withdrawals because we need a valid account proof for the
+///         L2ToL1MessagePasser via the top-level Monomer state root.
+contract L2ApplicationStateRootProvider {
+    /// @notice The current state root of the Cosmos application using Monomer.
+    uint256 public l2ApplicationStateRoot;
```

Comment:
In my head, hashed data like state roots are more commonly encoded `bytes32`. Equivalent data, but might save some reformatting work / issues in the future.

Reply:
Yeah that's a good point, I'll update this!